controllers/argocd/deployment.go

```diff
@@ -217,6 +235,14 @@ func getArgoRepoCommand(cr *argoprojv1a1.ArgoCD) []string {
 	cmd = append(cmd, "--redis")
 	cmd = append(cmd, getRedisServerAddress(cr))
 
+	if useTLSForRedis {
+		cmd = append(cmd, "--redis-use-tls")
+		cmd = append(cmd, "--redis-ca-certificate", "/app/config/reposerver/tls/redis/tls.crt")
+		if isRedisTLSVerificationDisabled(cr) {
+			cmd = append(cmd, "--redis-insecure-skip-tls-verify")
+		}
```

Comment:
I think the `--redis-ca-certificate` argument and the `--redis-insecure-skip-tls-verify` should be mutually exclusive, e.g. in two different branches of the same `if/else` block. Would that make sense?